src/cow-react/modules/limitOrders/pure/ReceiptModal/index.tsx

```diff
@@ -0,0 +1,133 @@
+import { GpModal } from 'components/Modal'
+import { CurrencyAmount, Fraction, Token } from '@uniswap/sdk-core'
+import * as styledEl from './styled'
+import { OrderKind } from 'state/orders/actions'
+import { CloseIcon } from 'theme'
+import { CurrencyField } from './CurrencyField'
+import { SupportedChainId } from '@cowprotocol/cow-sdk'
+import { ParsedOrder } from '@cow/modules/limitOrders/containers/OrdersWidget/hooks/useLimitOrdersList'
+import { FeeField } from './FeeField'
+import { StyledScrollarea } from 'components/SearchModal/CommonBases/CommonBasesMod'
+import { FieldLabel } from './FieldLabel'
+import { PriceField } from './PriceField'
+import { DateField } from './DateField'
+import { FilledField } from './FilledField'
+import { SurplusField } from './SurplusField'
+import { OrderIDField } from './OrderIdField'
+import { StatusField } from './StatusField'
+
+interface ReceiptProps {
+  isOpen: boolean
+  order: ParsedOrder
+  chainId: SupportedChainId
+  onDismiss: () => void
+  sellAmount: CurrencyAmount<Token>
+  buyAmount: CurrencyAmount<Token>
+  limitPrice: Fraction | null
+  executionPrice: Fraction | null
+}
+
+enum Tooltip {
+  STATUS = 'Status info TODO',
+  LIMIT_PRICE = 'Limit price TODO',
+  EXECUTION_PRICE = 'Execution price TODO',
+  FILLED = 'Filled TODO',
+  SURPLUS = 'Order Surplus TODO',
+  FEE = 'Fee TODO',
+  CREATED = 'Created date TODO',
+  EXPIRY = 'Expiry date TODO',
+  ORDER_TYPE = 'Order type TODO',
+  ORDER_ID = 'Order id TODO',
+}
+
+export function ReceiptModal({
+  isOpen,
+  onDismiss,
+  order,
+  chainId,
+  sellAmount,
+  buyAmount,
+  limitPrice,
+  executionPrice,
+}: ReceiptProps) {
+  if (!order || !chainId) {
+    return null
+  }
+
+  const inputLabel = order.kind === OrderKind.SELL ? 'You sell' : 'You sell at most'
+  const outputLabel = order.kind === OrderKind.SELL ? 'Your receive at least' : 'You receive exactly'
+  const orderType = order.kind === OrderKind.SELL ? 'Sell order' : 'Buy order'
+
+  return (
+    <GpModal onDismiss={onDismiss} isOpen={isOpen}>
+      <styledEl.Wrapper>
+        <styledEl.Header>
+          <styledEl.Title>Order Receipt</styledEl.Title>
+          <CloseIcon onClick={() => onDismiss()} />
+        </styledEl.Header>
+
+        <StyledScrollarea>
+          <styledEl.Body>
+            <CurrencyField amount={sellAmount} token={order.inputToken} label={inputLabel} />
+            <CurrencyField amount={buyAmount} token={order.outputToken} label={outputLabel} />
+
+            <styledEl.Field border="rounded-top">
+              <FieldLabel label="Status" tooltip={Tooltip.STATUS} />
+              <StatusField order={order} />
+            </styledEl.Field>
+
+            <styledEl.Field>
+              <FieldLabel label="Limit price" tooltip={Tooltip.LIMIT_PRICE} />
+              <PriceField order={order} price={limitPrice} />
```

Comment:
U really like you splitted this into smaller components. Makes it easy to mantain!

Reply:
Yep, I've started with one one component but then it grew quickly so I refactored it.